site/lib/mergeHoldings.test.ts

```diff
@@ -0,0 +1,406 @@
+      carbonValue: "2",
+      change: 1,
+    },
+    {
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "12",
+      token: "NCT",
+      tokenAmount: "3",
+      carbonValue: "3",
+      change: 1,
+    },
+    {
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "14",
+      token: "NCT",
+      tokenAmount: "4",
+      carbonValue: "4",
+      change: 1,
+    },
+  ],
+];
+
+const expectedOutput = {
+  BCT: [
+    {
+      carbonValue: "1",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-BCT-1674777600",
+      timestamp: "1",
+      token: "BCT",
+      tokenAmount: "1",
+    },
+    {
+      carbonValue: "2",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-BCT-1674777600",
+      timestamp: "3",
+      token: "BCT",
+      tokenAmount: "2",
+    },
+    {
+      carbonValue: "4",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-BCT-1674777600",
+      timestamp: "4",
+      token: "BCT",
+      tokenAmount: "4",
+    },
+    {
+      carbonValue: "5",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-BCT-1674777600",
+      timestamp: "5",
+      token: "BCT",
+      tokenAmount: "5",
+    },
+    {
+      carbonValue: "6",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-BCT-1674777600",
+      timestamp: "6",
+      token: "BCT",
+      tokenAmount: "6",
+    },
+  ],
+  NCT: [
+    {
+      carbonValue: "1",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-NCT-1674777600",
+      timestamp: "7",
+      token: "NCT",
+      tokenAmount: "1",
+    },
+    {
+      carbonValue: "2",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "8",
+      token: "NCT",
+      tokenAmount: "2",
+    },
+    {
+      carbonValue: "3",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-NCT-1674777600",
+      timestamp: "9",
+      token: "NCT",
+      tokenAmount: "3",
+    },
+    {
+      carbonValue: "4",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "10",
+      token: "NCT",
+      tokenAmount: "4",
+    },
+    {
+      carbonValue: "5",
+      id: "0xdeb8c24ad9640d62334da6f30c191e9cfd180e5b-NCT-1674777600",
+      timestamp: "11",
+      token: "NCT",
+      tokenAmount: "5",
+    },
+    {
+      carbonValue: "6",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "12",
+      token: "NCT",
+      tokenAmount: "6",
+    },
+    {
+      carbonValue: "7",
+      id: "0x838cd36c38e64c166b2c5130d15a8acff1e09d44-NCT-1674777600",
+      timestamp: "14",
+      token: "NCT",
+      tokenAmount: "7",
+    },
+  ],
+};
+
+export const holdingsTest = test("setup", () => {
+  expect(groupHoldings(holdingsInput)).toStrictEqual(groupedHoldings);
+  expect(calculateChange(groupedHoldings)).toStrictEqual(
+    calculatedGroupedHoldings
+  );
+  expect(calculateTokenValues(calculatedGroupedHoldings)).toStrictEqual(
```

Comment:
* [ ] we need a test for this util for balances that go down

* [ ] also I'd like to make sure it doesn't explode with big decimals. BCT has 18 decimals which can cause overflows and weirdness. So try doing another test with a fixture that has values like:
`0.111111111111111111`
plus `0.111111111111111111`
equals `0.222222222222222222`
if it causes an overflow diget thats ok, because its only used in the chart, we can at least document the quirky math with the test. or if you want to get fancy use `ethers.BigNumber.from(value).sub(othervalue).format()` to do the math...

Reply:
@Atmosfearful the output of those numbers is 16 digits so I created a test tase to document.